app/src/main/cpp/WidgetMover.cpp

```diff
@@ -162,13 +162,32 @@ WidgetMover::HandleMove(const vrb::Vector& aStart, const vrb::Vector& aDirection
     return m.HandleKeyboardMove(delta);
   } else {
     // General case
-    vrb::Matrix updatedTransform = m.initialTransform.Translate(vrb::Vector(delta.x(), delta.y(), 0.0f));
-    m.widget->SetTransform(updatedTransform);
-    m.endDelta.x() = delta.x() / WidgetPlacement::kWorldDPIRatio;
-    m.endDelta.y() = delta.y() / WidgetPlacement::kWorldDPIRatio;
-    m.endDelta.z() = 0.0f;
+    const float maxX = 1.5f;
+    const float minX = -maxX;
+    const float maxY = 3.0f;
+    const float minY = -maxY;
+    vrb::Vector translation = m.initialPlacement->translation;
+    float x = translation.x() * WidgetPlacement::kWorldDPIRatio;
+    float y = translation.y() * WidgetPlacement::kWorldDPIRatio;
+    x += delta.x();
+    y += delta.y();
```

Comment:
Why not doing these operations in the initialization ?

Reply:
Updated along with the code in `HandleKeyboardMove()` as the code you mentioned was copied there.